src/utils/propertyTree.ts

```diff
@@ -0,0 +1,89 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+export enum Container {
+    Object, 
+    Array
+}
+
+export class PropertyTree {
+    propertyName: string;
+    beginningLineNumber: number | undefined;
+    endLineNumber: number | undefined;
+    parent : PropertyTree | undefined;
+    commaIndex  : number | undefined;
+    commaLine : number | undefined;
+    lineWhereToAddComma : number | undefined;
+    indexWhereToAddComa : number | undefined;
+    type : Container | undefined;
+    childrenProperties: PropertyTree[];
+    lastProperty : boolean;
+    noKeyName : boolean;
+
+    constructor(
+        propertyName?: string, 
+        beginningLineNumber?: number
+        ) {
+
+        this.propertyName = propertyName ? propertyName : '';
```

Comment:
Note that `propertyName` is also true for empty string. It's better to write `propertyName !== undefined`. Or use the `??operator` which tests for undefined: `this.propertyName = propertyName ?? '';`